### PR TITLE
build: add workflow preferring TypeScript files

### PR DESCRIPTION
**Pull request by etr2460**
### SUMMARY
Adds a Github workflow to check if any `.js` or `.jsx` files were added in a PR, and adds a comment to the PR asking to change the files to use TypeScript. This check always passes and will never prevent merge.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Comments when a `.js` file is added:
Screen Shot 2020-05-24 at 10 25 17 PM

Skips the comment step when no js files are added:
Screen Shot 2020-05-24 at 10 27 58 PM

### TEST PLAN
Add a `.js` file to this PR and see the message appear. See the comment step skip when no js is added. Also test the `jq` command locally to ensure it works will all file extensions

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

to: @ktmud @john-bodley @kristw @nytai @graceguo-supercat

### CODE REVIEW

**Comment by ktmud** [comment]
This is a great idea!

File: .github/workflows/prefer_typescript.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        id: check
+        run: |
+          echo ::set-output name=was_js_file_added::$(jq 'map(endswith(".js") or endswith(".jsx"))' ${HOME}/files_added.json | jq 'reduce .[] as $is_js (false; . or $is_js)')
+      - if: steps.check.outputs.was_js_file_added != 'false'
+        name: Comment about preferring TypeScript
+        uses: unsplash/comment-on-pr@master
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          msg: "It looks like your PR contains new `.js` or `.jsx` files. As decided in [SIP-36](https://github.com/apache/incubator-superset/issues/9101), all new files should be written in TypeScript. Please convert new JavaScript files to TypeScript and then re-request review."
```
Comment (ktmud) [inline .github/workflows/prefer_typescript.yml:28]: How about a title to make this more visible 

```
        with:
          msg: |
            ## WARNING: TypeScript Preferred

            It looks like your PR contains new `.js` or `.jsx` files. As decided in [SIP-36](https://github.com/apache/incubator-superset/issues/9101), all new files should be written in TypeScript.

            Please convert new JavaScript files to TypeScript (`.ts` or `tsx`) then re-request review.
```

---

## WARNING: Prefer TypeScript

It looks like your PR contains new `.js` or `.jsx` files. As decided in [SIP-36](https://github.com/apache/incubator-superset/issues/9101), all new files should be written in TypeScript. Please convert new JavaScript files to TypeScript then re-request review.

---

It'd also be nice if the comment could be deleted when new `js` was removed from PR.

Comment (etr2460) [reply .github/workflows/prefer_typescript.yml:28]: That's a good idea, I'll add a title.

I couldn't find a github action for deleting comments unfortunately, so I don't think there's an easy way to implement your second comment...

**Comment by etr2460** [comment]
ping @ktmud for rereview

**Review by ktmud — APPROVED** [review]

**Pull request merged**